demo/src/screens/componentScreens/TextFieldScreen.tsx

```diff
@@ -261,33 +268,50 @@ export default class TextFieldScreen extends Component {
   }
 
   onChangeIndexFieldStyle = (index: number) => {
-    this.setState({customPreset: index === 0 ? 'underline' : 'outline'});
+    this.setState({preset: index === 0 ? 'underline' : 'outline'});
+  };
+
+  getDynamicFieldStyle = (context, props) => {
```

Comment:
This behavior can be achieved by passing a color object with all states, any reason you are doing it like this?

Reply:
We don't have a prop to control the underline color with the ColorType... What are you referring to?